File: src/main/java/me/muksc/tacztweaks/mixin/accessor/ClipContextAccessor.java
package me.muksc.tacztweaks.mixin.accessor;

import net.minecraft.world.level.ClipContext;
import net.minecraft.world.phys.shapes.CollisionContext;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.gen.Accessor;

@Mixin(ClipContext.class)
public interface ClipContextAccessor {
    @Accessor("block")
    ClipContext.Block tacztweaks$getBlock();

    @Accessor("fluid")
    ClipContext.Fluid tacztweaks$getFluid();

    @Accessor("collisionContext")
    CollisionContext tacztweaks$getCollisionContext();
}

File: src/main/kotlin/me/muksc/tacztweaks/mixininterop/ClipContextAccessor.kt
package me.muksc.tacztweaks.mixininterop

import me.muksc.tacztweaks.mixin.accessor.ClipContextAccessor
import net.minecraft.world.level.ClipContext
import net.minecraft.world.phys.shapes.CollisionContext

inline val ClipContextAccessor.block: ClipContext.Block
    get() = `tacztweaks$getBlock`()

inline val ClipContextAccessor.fluid: ClipContext.Fluid
    get() = `tacztweaks$getFluid`()

inline val ClipContextAccessor.collisionContext: CollisionContext
    get() = `tacztweaks$getCollisionContext`()